lib/pbench/test/unit/server/test_datasets_metadata.py

```diff
@@ -30,7 +31,12 @@ def query_api(
             except DatasetNotFound:
                 dataset = ds_name  # Allow passing deliberately bad value
             if username:
-                token = self.token(client, server_config, username)
+                token = generate_token(
+                    username=username,
+                    pbench_client_roles=["ADMIN"]
+                    if username == admin_username
+                    else None,
+                )
```

Comment:
This logic is encapsulated in the `token` in `test_datasets_list.py` in a more readable (less "blackened") form. Rather than duplicating the "admin or not admin" logic across 3 crispy lines all over the place, can we just do it once?

Reply:
Or, can we use the `pbench_token` and/or `pbench_admin_token` fixture values?  Ditto line 81.